DataCollector/src/main/java/com/braintreepayments/api/datacollector/DataCollectorInternalRequest.kt

```diff
@@ -0,0 +1,62 @@
+package com.braintreepayments.api.datacollector
+
+import androidx.annotation.RestrictTo
+import kotlin.math.min
+
+/**
+ * Used to configuration the PayPalDataCollector request
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class DataCollectorInternalRequest(val hasUserLocationConsent: Boolean) {
```

Comment:
Is it possible to make this a data class?

Reply:
> Is it possible to make this a data class?

It would be confusing to make it a data class because a data class must have properties in it primary constructor and those primary constructor properties cannot have custom setters or getters. We have one property clientMetadataId that requires custom setter and can be declared in the body only.

Reply:
I was thinking the data class features(equals() and hashCode()) by adding all the properties to its primary constructor, but since we are not going to compare the objects and the class is not Parcelable, probably it's fine to make DataCollectorInternalRequest a data class. I updated it already.

Reply:
Yeah we aren't using those InternalRequest objects as keys in a Map or anything so it should be fine. I've been leaning toward data classes more in our SDK because it seems like the value objects have too much responsibility at times e.g. `toJson()`. Ideally in our SDK we should have a set of pure data types and classes that leverage those data types to make HTTP requests.

Reply:
I agree, I also like the design of data class (a POJO with auto-generated equals) and the business logic could be implemented on the data consumer. When looking at the Google Pay tutorial(https://developers.google.com/pay/api/android/guides/tutorial), they also use JSONObject, JSONArray to wrap data(other modules are using those, for example, Card). I wonder maybe it's more suitable for payment api, sdk implementation, or the tutorial was written a while back.

Reply:
Yeah they switched to JSONObject a while back. For our SDK I think it makes sense to have strongly typed objects as a way to interact with our API. One of the main responsibilities of our Core SDK is to provide an interface to merchants to make payments that doesn't require them to think about Networking/HTTP request logic.

Reply:
That could be the next major refactoring after all the Kotlin conversion.